### PR TITLE
Relax `NoCell` bound of `TryFromBytes::try_from_mut`

### LINKED ISSUE #1014: Relax `NoCell` bound on `Ptr::try_cast_into`
Follow-up to #999.

### PR DESCRIPTION
**Pull request by jswrenn**
Makes progress towards #5.

Closes #1014.

### CODE REVIEW

File: src/pointer/ptr.rs [hunk 1]
```diff
@@ -1464,6 +1469,45 @@ mod _project {
...
 }
 
+/// Implemented if `Ptr<T>` can be casted to `Ptr<U>` without violating the
+/// `UnsafeCell`-related invariants of [`Ptr`].
+///
+/// # Safety
+///
+/// `Ptr` requires that:
+///
+/// > During the lifetime 'a, no live reference or live `Ptr` will exist to this
```
Comment (joshlf) [inline src/pointer/ptr.rs:1479]:
```suggestion
/// > During the lifetime `'a`, no live reference or live `Ptr` will exist to this
```

**Review by joshlf — CHANGES_REQUESTED** [review]

File: src/pointer/ptr.rs (hunk 1 above)
Comment (jswrenn) [reply src/pointer/ptr.rs]: Done (this particular doc comment no longer exists).

File: src/pointer/ptr.rs [hunk 2]
```diff
@@ -1464,6 +1469,45 @@ mod _project {
...
+/// > memory which treats `UnsafeCell`s as existing at different ranges than
+/// > they exist in `T`.
+///
+/// Implementations of this trait must prove that for an aliasing `A`, a cast of
```
[the remark this answers is not in the document]
Comment (jswrenn) [reply src/pointer/ptr.rs]: Done (this particular doc comment no longer exists).

File: src/pointer/ptr.rs [hunk 3]
```diff
@@ -1464,6 +1469,45 @@ mod _project {
...
+/// `Ptr<T, Aliasing = A>` to `Ptr<U, Aliasing = A>` does not introduce a live
+/// reference to `U` with `UnsafeCell`s at different ranges than a live
```
[the remark this answers is not in the document]
Comment (jswrenn) [reply src/pointer/ptr.rs]: Done (this particular doc comment no longer exists).

File: src/pointer/ptr.rs [hunk 4]
```diff
@@ -1464,6 +1469,45 @@ mod _project {
...
+/// reference to `T`.
```
[the remark this answers is not in the document]
Comment (jswrenn) [reply src/pointer/ptr.rs]: Done (this particular doc comment no longer exists).

**Pull request closed without merging**